integration/graphql/src/main/java/com/blazebit/persistence/integration/graphql/GraphQLEntityViewSupport.java

```diff
@@ -389,48 +388,25 @@ public <T> EntityViewSetting<T, CriteriaBuilder<T>> createSetting(DataFetchingEn
     }
 
     public String getElementTypeName(DataFetchingEnvironment dataFetchingEnvironment, String elementRoot) {
-        GraphQLType type = getElementType(dataFetchingEnvironment, elementRoot);
-        if (type instanceof GraphQLObjectType) {
-            return ((GraphQLObjectType) type).getName();
-        } else {
-            return ((GraphQLInterfaceType) type).getName();
-        }
+        return getElementType(dataFetchingEnvironment, elementRoot).getName();
     }
 
-    public GraphQLType getElementType(DataFetchingEnvironment dataFetchingEnvironment, String elementRoot) {
-        GraphQLType type = dataFetchingEnvironment.getFieldDefinition().getType();
+    public GraphQLUnmodifiedType getElementType(DataFetchingEnvironment dataFetchingEnvironment, String elementRoot) {
```

Comment:
It is actually compatible because the other types (not GraphQLUnmodifiedType) were precisely those which were manually unwrapped (List and NoNull). This is now done via the `unwrapAll` util function.